File: src/main/java/de/spries/fleetcommander/service/core/dto/GameAccessParams.java
package de.spries.fleetcommander.service.core.dto;

import com.fasterxml.jackson.annotation.JsonIgnore;

public class GameAccessParams {

	private GamePlayer gamePlayer;
	private String authToken;

	public GameAccessParams(GamePlayer gamePlayer, String authToken) {
		this.gamePlayer = gamePlayer;
		this.authToken = authToken;
	}

	public int getGameId() {
		return gamePlayer.getGameId();
	}

	@JsonIgnore
	public int getPlayerId() {
		return gamePlayer.getPlayerId();
	}

	@JsonIgnore
	public String getAuthToken() {
		return authToken;
	}

	public String getFullAuthToken() {
		return getPlayerId() + ":" + authToken;
	}

}

File: src/main/kotlin/de/spries/fleetcommander/service/core/dto/GameAccessParams.kt
package de.spries.fleetcommander.service.core.dto

import com.fasterxml.jackson.annotation.JsonIgnore

data class GameAccessParams(@get:JsonIgnore
                            val gamePlayer: GamePlayer,
                            @get:JsonIgnore
                            val authToken: String) {

    fun getGameId() = gamePlayer.gameId

    @JsonIgnore
    fun getPlayerId() = gamePlayer.playerId

    fun getFullAuthToken() = "${getPlayerId()}:$authToken"

}
